### PR TITLE
Basic implementation of ls

### PR DESCRIPTION
**Pull request by peschkaj**
This is a basic implementation of `ls` that produces output roughly analogous to `ls -l`. The `-h`/`--human-readable` option is also implemented.

See #126 for outstanding items.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -145,6 +146,7 @@ TEST_PROGS  := \
 	head \
 	link \
 	ln \
+    ls \
```
Comment (Heather) [inline Makefile]: tabs / spaces

Comment (peschkaj) [inline Makefile]: Fixed this. Will fix failing TravisCI tests shortly.

File: src/ls/ls.rs [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+#[allow(unused_variables)]
+fn display_permissions(metadata: &Metadata) -> String {
+    String::from("---------")
+}
+
+#[cfg(target_family = "unix")]
+fn display_permissions(metadata: &Metadata) -> String {
+    use std::os::unix::fs::PermissionsExt;
+    
+    "xxxxxxxxx".to_string();
```
Comment (jbcrail) [inline src/ls/ls.rs]: The trailing comma causes the compiler to fail.

Comment (peschkaj) [inline src/ls/ls.rs]: Yeah, conditional compilation bit me on this. Latest commit removes this and a local VM will prevent it going forward. Apologies for the noise on the repo.

**Comment by peschkaj** [comment]
Looks like the integration test `test_integration_pwd` is failing.

**Comment by jbcrail** [comment]
Yeah, the Windows port is still a work-in-progress. You can ignore the errors for now. PR #853 will take care of `pwd`, but there are other failed tests after that.

**Comment by peschkaj** [comment]
@jbcrail - I don't see any failures specifically for `ls`. Did I miss something? (Not being snarky, just sorting through AppVeyor's output).

I just want to make sure I'm not blocking progress on this.

**Comment by Heather** [comment]
@peschkaj thanks for that work, future improvements are very welcome

**Comment by peschkaj** [comment]
My pleasure! I hope to add more over time.

On Sat, Apr 30, 2016 at 4:56 PM Heather <notifications@github.com> wrote:

> @peschkaj <https://github.com/peschkaj> thanks for that work, future
> improvements are very welcome
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/uutils/coreutils/pull/845#issuecomment-215994960>
>
-- 

Jeremiah Peschka

**Comment by gokussjx** [comment]
Thanks for this!
Would you mind adding it to the README as well?
https://github.com/uutils/coreutils/blob/master/README.md#utilities

**Pull request merged**